### DISCUSSION TITLE
How to position shapes (triangle)

### DISCUSSION THREAD

**Discussion by hamzap314** [3b1b/manim#1308] [Q&A] [answered]
Hello, 
I hope you all are doing well. 
I am completely new to manim and I started looking at it a few days ago. 
I was wondering how the positioning in manim works. 
So here is my code:
```
triangle_big = Triangle()
triangle_small = Triangle()
triangle_small.scale(0.5)
self.play(ShowCreation(triangle_big))
self.wait()
self.play(ShowCreation(triangle_small))
self.wait()
```
This puts the triangles on top of each other like this:
![image](https://user-images.githubusercontent.com/55862742/104068578-d0027e00-51ca-11eb-9b2a-753dbff3b91e.png)
I want the triangles to be like this, with one of the vertices of each triangle aligned:
![image](https://user-images.githubusercontent.com/55862742/104069933-59b24b80-51cb-11eb-8512-3806f47eb82a.png)
Could somebody please show me how to do this?
Thank you so much. 
Please let me know if my question is in the wrong place, or needs more clarification.
- Hamza

**Comment by TonyCrane** [comment, marked as answer]
Sorry to respond so late.
Is this what you want?
![image](https://user-images.githubusercontent.com/44120331/104123953-fec65300-5388-11eb-8453-8602a6637cb1.png)

The best way to do this is to use `align_to` method:
```python
triangle_small.align_to(triangle_big, DL)
```
and you can also use `next_to` or `shift`:
```python
triangle_small.next_to(triangle_big.get_vertices()[1], aligned_edge=DL, buff=0)
triangle_small.shift(triangle_big.get_vertices()[1] - triangle_small.get_vertices()[1])
```
`get_vertices` method get the vertices (a list of points) of the triangle, and the indexes of the list of vertices are:
![image](https://user-images.githubusercontent.com/44120331/104124050-7ac09b00-5389-11eb-82b3-4d76df363d8c.png)
So the vertice in the lower left corner is `triangle.get_vertices()[1]`.
And you can also directly put the small triangle to the target position by using the `about_point` parameter of `scale`:
```python
triangle_small.scale(0.5, about_point=triangle_big.get_vertices()[1])
```
---
btw, if you want to learn more about position in manim, you can watch [this video](https://www.bilibili.com/video/BV1p54y197cC). (Although this video is in Chinese, it has no great influence on learning the content. You can just watch the screen and ignore the Chinese subtitles.)

**Reply by hamzap314** [reply]
Thank you so much for such a thorough answer! I really appreciate your help Tony. Have a great day.

**Reply by 3b1b** [reply]
You can also use an argument of move_to, for example

`triangle_small.move_to(triangle_big, DL)`
